tests/stdlib/tunittest.nim

```diff
@@ -120,3 +122,31 @@ suite "bug #5784":
       field: int
     var obj: Obj
     check obj.isNil or obj.field == 0
+
+when defined(testing):
+  suite "test name filtering":
+    test "test name":
+      check matchFilter("suite1", "foo", "")
+      check matchFilter("suite1", "foo", "foo")
+      check matchFilter("suite1", "foo", "::")
+      check matchFilter("suite1", "foo", "::foo")
```

Comment:
```nim
check matchFilter("suite1", "::foo", "::foo")
check matchFilter("suite1", "qwe::foo", "qwe::foo")
check matchFilter("suite1", "qwe::foo", "suite1::qwe::foo")
check matchFilter("suite1", "q**we::foo", "q**we::foo")
check matchFilter("suite1", "q*wefoo", "q*wefoo")
```

I think we should test this as well. For tests that have ``*`` in the name, a verbatim name should match.